app/controllers/api/users.php

```diff
@@ -1156,11 +1156,19 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
             /** @var Document $session */
             $dbForProject->deleteDocument('sessions', $session->getId());
             //TODO: fix this
+
+            $queueForEvents
+                ->setEvent('users.[userId].sessions.[sessionId].delete')
+                ->setParam('userId', $user->getId())
+                ->setParam('sessionId', $session->getId())
+                ->setPayload($response->output($session, Response::MODEL_SESSION))
+                ->trigger();
         }
 
         $dbForProject->deleteCachedDocument('users', $user->getId());
 
         $queueForEvents
+            ->setEvent('users.[userId].sessions.delete')
```

Comment:
We won't need this here - it will be set from the routes `event` label in [the shutdown hook](https://github.com/appwrite/appwrite/blob/5871178fc0bbc6edfcba6fe39cce3d2f3160a884/app/controllers/shared/api.php#L177)

Reply:
Removed

Reply:
Had to add it back, that hook is run before each request no? `init`

Reply:
Ah my bad, I thought it was in a shutdown hook. It probably should be, it's not really of any use before the route action runs 🤔 that's outside the scope of this PR though